src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -76,38 +89,90 @@ protected void start(Map<String, String> props, ElasticsearchClient client) {
       log.warn("AK versions prior to 2.6 do not support the errant record reporter.");
     }
 
-    this.client = client != null ? client : new ElasticsearchClient(config, reporter);
+    this.client = client != null ? client
+            : new ElasticsearchClient(config, reporter, offsetTracker);
 
     log.info("Started ElasticsearchSinkTask. Connecting to ES server version: {}",
         getServerVersion());
   }
 
+  private static class PartitionPauser {
+
+    private final SinkTaskContext context;
+    private final BooleanSupplier pauseCondition;
+    private final BooleanSupplier resumeCondition;
+    private boolean partitionsPaused;
+
+    public PartitionPauser(SinkTaskContext context,
+                           BooleanSupplier pauseCondition,
+                           BooleanSupplier resumeCondition) {
+      this.context = context;
+      this.pauseCondition = pauseCondition;
+      this.resumeCondition = resumeCondition;
+    }
+
+    /**
+     * Resume partitions if they are paused and resume condition is met.
+     * Has to be run in the connector thread.
+     */
+    private void tryResumePartitions() {
```

Comment:
Would be good to add unit tests for this class.

Reply:
Added